src/client.js

```diff
@@ -67,15 +67,15 @@ function Client(manager, name, config) {
 
 	var client = this;
 
-	if (config) {
-		if (!config.token) {
-			client.updateToken(function() {
-				client.manager.updateUser(client.name, {token: config.token});
+	if (client.config) {
```

Comment:
This line is a bit suspect, I think we should default `config` to `{}` if it's undefined, and remove the check (I saw another `|| {}` in the code.